### DISCUSSION TITLE
Missing images in document with duplicated images

### DISCUSSION THREAD

**Discussion by jnsebgosselin** [py-pdf/pypdf#2472] [Q&A] [answered]
I want to create a microsoft word document and use a placeholder image that I want to replace with the proper images with pypdf after the document has been converted to pdf.

I have created the attached document with microsoft word using the same image three times (3X). However, when I extract the images with `pypdf` only one image is listed. I think microsoft is now checking for duplicate images to optimize space (when I look into the docx document with 7zip, only one image is stored in there).

Is there something that can be done do differentiate these images with pypdf other than using distinct images for the placeholder?

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Windows-10-10.0.19045-SP0

$ python -c "import pypdf;print(pypdf._debug_versions)"
pypdf==4.0.2, crypt_provider=('cryptography', '42.0.1'), PIL=8.2.0
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from pypdf import PdfReader
pdf_fpath = 'placeholder.pdf'
reader = PdfReader(pdf_fpath)
page = reader.pages[0]
images = page.images    
print(len(images))
```

[placeholder.pdf](https://github.com/py-pdf/pypdf/files/14405941/placeholder.pdf)

**Comment by pubpub-zz** [comment, marked as answer]
Images property extracts the images stored attached to the page. As noticed If images are called multiple times they only appear once.
In order to view the calls you have to parse the content. The easiest is to get it as operations and look for Do operations or BI for inline images.
2 warnings:
Do operations 'calls' images but also sub drawings: you have to ensure it is an image type object
Some images are included in sub drawings.

**Comment by jnsebgosselin** [comment]
Thanks for the help @pubpub-zz !

And if I locate the calls to the image, will this be feasible to remove those calls and insert another image in their place?

**Reply by pubpub-zz** [reply]
If you mean use the `.replace()` function this will replace the internal object and will display the image in all positions.

**Reply by jnsebgosselin** [reply]
No not exactly. What I would like to do is to replace each call to the internal object (the placeholder image) with a different image.

Anyway, even though this is not what I was hoping for, I understand now that what I want to do will require a lot more work thank expected.

Thank you again for you help. I will mark this as resolved.
